diff --git a/src/swig/mlt.i b/src/swig/mlt.i
index ca0b58a1..0644c84b 100644
--- a/src/swig/mlt.i
+++ b/src/swig/mlt.i
@@ -240,7 +240,11 @@ binary_data frame_get_image( Mlt::Frame &frame, mlt_image_format format, int w,
 %}
 
 %typemap(out) binary_data {
-	$result = PyString_FromStringAndSize( $1.data, $1.size );
+	%#if PY_MAJOR_VERSION < 3
+		$result = PyString_FromStringAndSize( $1.data, $1.size );
+	%#else
+		$result = PyUnicode_FromStringAndSize( $1.data, $1.size );
+	%#endif
 }
 
 binary_data frame_get_waveform(Mlt::Frame&, int, int);
